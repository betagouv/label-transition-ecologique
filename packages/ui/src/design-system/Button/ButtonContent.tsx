import {Icon, IconSize} from '@design-system/Icon';
import {buttonThemeClassnames} from './theme';
import {ButtonContentProps, ButtonSize} from './types';

const getIconSize = (size: ButtonSize): IconSize | undefined => {
  // Les sizes des icÃ´nes ne matchent pas celles des boutons
  switch (size) {
    case 'xs':
      return 'sm';
    case 'sm':
      return 'md';
    case 'md':
      return 'md';
    case 'xl':
      return 'lg';
  }
};

/** Affiche le contenu d'un bouton */
const ButtonContent = ({
  variant = 'primary',
  size = 'md',
  icon,
  children,
  disabled,
}: ButtonContentProps) => {
  const buttonState = disabled ? 'disabled' : 'default';

  return (
    <>
      {!!icon && (
        <Icon
          icon={icon}
          size={getIconSize(size)}
          className={buttonThemeClassnames[variant][buttonState].icon}
        />
      )}

      {!!children && children}
    </>
  );
};

export default ButtonContent;
