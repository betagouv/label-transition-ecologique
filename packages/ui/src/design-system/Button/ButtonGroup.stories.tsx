import {useState} from 'react';
import {Meta, StoryObj} from '@storybook/react';

import {ButtonGroup} from './ButtonGroup';

const meta: Meta<typeof ButtonGroup> = {
  component: ButtonGroup,
  args: {
    activeButtonId: '2',
    buttons: [
      {id: '1', children: 'Bouton 1'},
      {id: '2', children: 'Bouton 2'},
      {id: '3', children: 'Bouton 3'},
    ],
  },
  render: args => {
    const [activeId, setActiveId] = useState(args.activeButtonId);
    return (
      <ButtonGroup
        {...args}
        activeButtonId={activeId}
        buttons={args.buttons.map(button => ({
          ...button,
          onClick: () => setActiveId(button.id),
        }))}
      />
    );
  },
};

export default meta;

type Story = StoryObj<typeof ButtonGroup>;

export const Default: Story = {
  args: {},
};

export const Size: Story = {
  args: {
    size: 'xs',
  },
};

export const Disabled: Story = {
  args: {
    buttons: [
      {id: '1', children: 'Bouton 1', disabled: true},
      {id: '2', children: 'Bouton 2'},
      {id: '3', children: 'Bouton 3'},
    ],
  },
};

export const Deux: Story = {
  args: {
    buttons: [
      {id: '1', children: 'Bouton 1'},
      {id: '2', children: 'Bouton 2'},
    ],
  },
};
