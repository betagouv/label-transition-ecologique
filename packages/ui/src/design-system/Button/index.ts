export * from './Button';
export * from './NotificationButton';
