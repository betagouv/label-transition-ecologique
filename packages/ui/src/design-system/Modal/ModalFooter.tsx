import {Divider} from '@design-system/Divider';
import classNames from 'classnames';

type ModalFooterVariant = 'center' | 'right' | 'space';

const variantToClassnames: Record<ModalFooterVariant, string> = {
  center: 'justify-center',
  right: 'justify-end',
  space: 'justify-between',
};

export type ModalFooterProps = {
  /** Contenu principal du pied de page */
  children: React.ReactNode | React.ReactNode[];
  /** Contenu optionnel entre la ligne de séparation et le contenu principal */
  content?: React.ReactNode | React.ReactNode[];
  /** Variante visuelle */
  variant?: ModalFooterVariant;
};

/**
 * Composant de base pour le pied-de-page des modales.
 * Contient généralement un ou plusieurs boutons.
 */
export const ModalFooter = ({
  children,
  content,
  variant = 'right',
}: ModalFooterProps) => {
  return (
    <>
      <Divider className="mt-4" />
      {content}
      <div
        className={classNames(
          'flex gap-4 flex-wrap',
          variantToClassnames[variant]
        )}
      >
        {children}
      </div>
    </>
  );
};

/**
 * Conteneur lorsque le pied-de-page d'une modale doit être séparée en plusieurs
 * sections (variante `space`).
 */
export const ModalFooterSection = ({
  children,
}: {
  children: React.ReactNode | React.ReactNode[];
}) => {
  return <div className="flex gap-4 flex-wrap">{children}</div>;
};
