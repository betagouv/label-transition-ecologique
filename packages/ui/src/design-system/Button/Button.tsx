import {Ref, forwardRef} from 'react';
import classNames from 'classnames';

import {ButtonHTMLProps, isAnchor} from 'utils/types';
import {buttonSizeClassnames, buttonThemeClassnames} from './theme';
import ButtonContent from './ButtonContent';
import {ButtonContentProps, ButtonProps} from './types';

/**
 * Composant bouton par défaut, ayant pour props toutes les props habituelles d'un button tag.
 * */
// On déstructure toutes les props rajoutées qui ne sont pas des props des tags HTML <button> ou <a>
// Ce qui nous permet de ne donner que les props restantes natives au tag HTML
export const Button = forwardRef(
  (
    {
      children,
      disabled = false,
      variant = 'primary',
      size = 'md',
      className,
      icon,
      iconPosition,
      external,
      ...props
    }: ButtonProps,
    ref?: Ref<HTMLButtonElement | HTMLAnchorElement>
  ) => {
    const isIconButton = !children;

    const buttonState = disabled ? 'disabled' : 'default';

    const {text, background, border} =
      buttonThemeClassnames[variant][buttonState];

    const sizeClassName =
      buttonSizeClassnames[size][isIconButton ? 'iconButton' : 'textButton'];

    const buttonClassname = classNames(
      'w-fit flex items-center border-solid group',
      {
        // Layout du bouton
        'gap-1': size === 'xs' || (size === 'sm' && variant === 'underlined'),
        'gap-2':
          (size === 'sm' && variant !== 'underlined') ||
          size === 'md' ||
          size === 'xl',
        // Styles du curseur
        'cursor-pointer': !disabled,
        'cursor-not-allowed': disabled,
        // Bordures et polices
        'rounded-lg border font-bold': variant !== 'underlined',
        'border-b border-t-0 border-x-0 !p-px font-medium':
          variant === 'underlined',
        'hover:border-b-2 hover:!pb-0': variant === 'underlined' && !disabled,
      },
      text,
      background,
      border,
      sizeClassName
    );

    /** Reconstitution des props données au contenu du bouton */
    const buttonContentProps: ButtonContentProps = {
      children,
      variant,
      size,
      icon,
      disabled,
    };

    const isButton = !isAnchor(props);

    /** On affiche un bouton par défaut */
    if (isButton) {
      // On réintegre la prop disabled qui a été déstructurée plus haut
      const buttonProps = {...props, disabled} as ButtonHTMLProps;
      return (
        <button
          ref={ref as Ref<HTMLButtonElement>}
          {...buttonProps}
          className={classNames(
            buttonClassname,
            {'flex-row-reverse': iconPosition === 'right'},
            className
          )}
          onClick={evt => {
            evt.stopPropagation();
            buttonProps.onClick?.(evt);
          }}
        >
          <ButtonContent {...buttonContentProps} />
        </button>
      );

      /** Ou bien une ancre si un lien href est donné (cf ./types.ts) */
    } else {
      const anchorProps = props;
      const openInNewTab = external || props.target === '_blank';
      return (
        <a
          {...anchorProps}
          ref={ref as Ref<HTMLAnchorElement>}
          // bg-none permet d'effacer un style dsfr appliqué à la balise <a/>
          // after:hidden supprime l'icône external par défaut du dsfr
          className={classNames(
            'bg-none after:hidden',
            buttonClassname,
            {
              'flex-row-reverse':
                iconPosition === 'right' ||
                (openInNewTab && iconPosition !== 'left'),
            },
            className
          )}
          target={openInNewTab ? '_blank' : anchorProps.target}
          rel={openInNewTab ? 'noreferrer noopener' : anchorProps.rel}
          onClick={evt => {
            evt.stopPropagation();
            if (disabled) evt.preventDefault();
            else anchorProps.onClick?.(evt);
          }}
        >
          <ButtonContent
            {...buttonContentProps}
            icon={openInNewTab ? 'external-link-line' : icon}
          />
        </a>
      );
    }
  }
);
