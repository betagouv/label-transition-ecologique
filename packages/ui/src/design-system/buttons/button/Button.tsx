import classNames from 'classnames';
import {buttonThemeClassnames} from '../utils';
import ButtonContent from './ButtonContent';
import {ButtonProps, isAnchorButton, isDefaultButton} from '../types';

/**
 * Composant bouton par défaut, ayant pour props toutes les props habituelles d'un button tag.
 * */
export const Button = (props: ButtonProps) => {
  const {
    children,
    disabled = false,
    variant = 'primary',
    size = 'md',
    className,
  } = props;

  const isIconButton = !children;
  const {text, background, border} = buttonThemeClassnames[variant];

  const buttonClassname = classNames(
    'rounded-lg w-fit border border-solid font-bold group',
    {
      // Format du bouton en fonction de la taille choisie
      'text-xs py-2': size === 'xs',
      'text-sm py-3': size === 'sm',
      'text-base py-3.5': size === 'md',
      'text-2xl py-5': size === 'xl',

      // Padding en fonction de la taille choisie
      'px-5':
        (size === 'xs' && !isIconButton) || (size === 'xl' && isIconButton),
      'px-6': size !== 'xs' && !isIconButton,

      // Padding sur les boutons ne contenant qu'une icône
      'px-2': size === 'xs' && isIconButton,
      'px-3': size === 'sm' && isIconButton,
      'px-3.5': size === 'md' && isIconButton,

      // Styles du curseur
      'cursor-pointer': !disabled,
      'cursor-not-allowed': disabled,
    },
    text,
    background,
    border
  );

  if (isDefaultButton(props)) {
    return (
      <button {...props} className={classNames(buttonClassname, className)}>
        <ButtonContent {...{...props, children, disabled, variant, size}} />
      </button>
    );
  } else if (isAnchorButton(props)) {
    const {external = false} = props;

    return (
      <a
        {...props}
        // bg-none permet d'effacer un style dsfr appliqué à la balise <a/>
        // after:hidden supprime l'icône external par défaut du dsfr
        className={classNames(
          'inline-block w-fit bg-none after:hidden',
          buttonClassname,
          className
        )}
        target={external ? '_blank' : props.target}
        rel={external ? 'noreferrer noopener' : props.rel}
        onClick={evt => {
          if (disabled) evt.preventDefault();
        }}
      >
        <ButtonContent
          {...{...props, children, disabled, variant, size}}
          external={external || props.target === '_blank'}
        />
      </a>
    );
  }
};
