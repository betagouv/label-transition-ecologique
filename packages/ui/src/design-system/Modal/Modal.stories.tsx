import {Meta, StoryObj} from '@storybook/react';
import {forwardRef} from 'react';

import {Modal} from '.';
import {Button} from '@design-system/Button';
import {ModalFooter} from '@design-system/Modal/ModalFooter';

const meta: Meta<typeof Modal> = {
  component: Modal,
  decorators: [story => <div className="p-8">{story()}</div>],
};

export default meta;

type Story = StoryObj<typeof Modal>;

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const OpenButton = forwardRef((props: any, ref) => (
  <Button {...props} ref={ref} variant="outlined">
    {props.text ? props.text : 'Open'}
  </Button>
));

export const Default: Story = {
  args: {
    textAlign: 'left',
    title: 'Un titre simple',
    subTitle: 'Un sous titre',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, semper eget tortor sed, aliquet ornare risus. Sed egestas egestas porttitor. Sed quis pretium eros. Mauris a turpis eu elit efficitur vehicula. Nulla ac vulputate velit. Nulla quis neque nec sapien molestie imperdiet. Cras viverra lacus vulputate diam malesuada viverra.',
  },
  render: args => {
    return (
      <Modal {...args}>
        <OpenButton text="Ouvrir la modale" />
      </Modal>
    );
  },
};

export const AvecFooter: Story = {
  args: {
    textAlign: 'left',
    title: 'Un titre simple',
    subTitle: 'Un sous titre',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, semper eget tortor sed, aliquet ornare risus. Sed egestas egestas porttitor. Sed quis pretium eros. Mauris a turpis eu elit efficitur vehicula. Nulla ac vulputate velit. Nulla quis neque nec sapien molestie imperdiet. Cras viverra lacus vulputate diam malesuada viverra.',
    renderFooter: ({close}) => (
      <ModalFooter>
        <Button onClick={close}>Fermer</Button>
      </ModalFooter>
    ),
  },
  render: args => {
    return (
      <Modal {...args}>
        <OpenButton text="Ouvrir la modale" />
      </Modal>
    );
  },
};

export const LongContent: Story = {
  args: {
    textAlign: 'left',
    title: 'Un titre simple',
    subTitle: 'Un sous titre',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, semper eget tortor sed, aliquet ornare risus. Sed egestas egestas porttitor. Sed quis pretium eros. Mauris a turpis eu elit efficitur vehicula. Nulla ac vulputate velit. Nulla quis neque nec sapien molestie imperdiet. Cras viverra lacus vulputate diam malesuada viverra. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, semper eget tortor sed, aliquet ornare risus. Sed egestas egestas porttitor. Sed quis pretium eros. Mauris a turpis eu elit efficitur vehicula. Nulla ac vulputate velit. Nulla quis neque nec sapien molestie imperdiet. Cras viverra lacus vulputate diam malesuada viverra. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, semper eget tortor sed, aliquet ornare risus. Sed egestas egestas porttitor. Sed quis pretium eros. Mauris a turpis eu elit efficitur vehicula. Nulla ac vulputate velit. Nulla quis neque nec sapien molestie imperdiet. Cras viverra lacus vulputate diam malesuada viverra. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, semper eget tortor sed, aliquet ornare risus. Sed egestas egestas porttitor. Sed quis pretium eros. Mauris a turpis eu elit efficitur vehicula. Nulla ac vulputate velit. Nulla quis neque nec sapien molestie imperdiet. Cras viverra lacus vulputate diam malesuada viverra. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, semper eget tortor sed, aliquet ornare risus. Sed egestas egestas porttitor. Sed quis pretium eros. Mauris a turpis eu elit efficitur vehicula. Nulla ac vulputate velit. Nulla quis neque nec sapien molestie imperdiet. Cras viverra lacus vulputate diam malesuada viverra.',
  },
  render: args => {
    return (
      <Modal
        {...args}
        render={({close}) => (
          <div className="flex flex-col p-8 border border-grey-5 text-grey-8 rounded-lg">
            <p>Contenu de la fonction "render" ici avec un petit bouton.</p>
            <p>Un autre paragraphe.</p>
            <div className="flex gap-6 mt-2 ml-auto">
              <Button variant="grey" onClick={() => close()}>
                Annuler
              </Button>
              <Button>Valider</Button>
            </div>
          </div>
        )}
      >
        <OpenButton text="Ouvrir la modale" />
      </Modal>
    );
  },
};

export const WithRender: Story = {
  args: {
    title: 'Un titre simple',
    subTitle: 'Un sous titre',
  },
  render: args => {
    return (
      <Modal
        {...args}
        render={() => (
          <div className="flex flex-col p-8 border border-grey-5 text-grey-8 rounded-lg">
            <p>Contenu de la fonction "render" ici avec un petit bouton.</p>
            <p>Un autre paragraphe.</p>
            <Button className="ml-auto">Valider</Button>
          </div>
        )}
      >
        <OpenButton text="Ouvrir la modale" />
      </Modal>
    );
  },
};

export const onlyRender: Story = {
  args: {noCloseButton: true},
  render: args => {
    return (
      <Modal
        {...args}
        render={() => (
          <div className="p-8 border border-grey-5 text-grey-8 rounded-lg">
            <span>Contenu de la fonction "render" ici dans les bordures.</span>
          </div>
        )}
      >
        <OpenButton text="Ouvrir la modale" />
      </Modal>
    );
  },
};

export const Size: Story = {
  args: {
    textAlign: 'left',
    title: 'Un titre simple',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, semper eget tortor sed, aliquet ornare risus. Sed egestas egestas porttitor. Sed quis pretium eros. Mauris a turpis eu elit efficitur vehicula. Nulla ac vulputate velit. Nulla quis neque nec sapien molestie imperdiet. Cras viverra lacus vulputate diam malesuada viverra.',
  },
  render: args => {
    return (
      <div className="flex items-center gap-6">
        <Modal {...args} size="sm">
          <OpenButton text="sm" />
        </Modal>
        <Modal {...args} size="md">
          <OpenButton text="md" />
        </Modal>
        <Modal {...args} size="lg">
          <OpenButton text="lg" />
        </Modal>
        <Modal {...args} size="xl">
          <OpenButton text="xl" />
        </Modal>
      </div>
    );
  },
};
