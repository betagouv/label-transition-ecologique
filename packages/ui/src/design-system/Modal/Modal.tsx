import {cloneElement, useState} from 'react';
import {
  FloatingFocusManager,
  FloatingOverlay,
  FloatingPortal,
  useClick,
  useDismiss,
  useFloating,
  useId,
  useInteractions,
  useRole,
} from '@floating-ui/react';
import classNames from 'classnames';

import {preset} from '@tailwind-preset';
import {Button} from '../Button';

type Size = 'sm' | 'md' | 'lg' | 'xl';

const sizeToClass: Record<Size, string> = {
  sm: 'max-w-sm',
  md: 'max-w-xl',
  lg: 'max-w-4xl',
  xl: 'max-w-[1200px]',
};

/**
 * Props données à la fonction qui rend le contenu de la modale,
 * accessibles dans les composants utilisant la modale
 */
export type RenderProps = {
  /** permet de fermer la modale */
  close: () => void;
  /** l'id à donner au titre pour l'accessibilité */
  labelId: string;
  /** l'id à la description pour l'accessibilité */
  descriptionId: string;
};

/** Types des props du composant générique Modal */
export type ModalProps = {
  /** Fonction de rendu du contenu de la modale */
  render?: (props: RenderProps) => React.ReactNode;
  /** Rendu pied de page optionnel */
  renderFooter?: (props: Pick<RenderProps, 'close'>) => React.ReactNode;
  /** l'élément qui permet d'afficher la modale au click, généralement un bouton */
  children?: JSX.Element;
  /** Titre de la modale, n'est pas affiché si non défini */
  title?: string;
  /** Sous-titre de la modale, n'est pas affiché si non défini */
  subTitle?: string;
  /** Description de la modale, n'est pas affichée si non définie. Elle est placée sous le titre */
  description?: string;
  /** Titre et description centrés par défaut */
  textAlign?: 'left' | 'center' | 'right';
  /** Permet de contrôler l'ouverture de la modale */
  openState?: {
    /** état d'ouverture de la modale */
    isOpen: boolean;
    /* accompagne "isOpen" afin de pouvoir fermer la modale */
    setIsOpen: (opened: boolean) => void;
  };
  /** fonction appelée lors de la fermeture de la modale */
  onClose?: () => void;
  /** max-width prédéfinies dans le DSFR, valeur par défaut "md" */
  size?: 'sm' | 'md' | 'lg' | 'xl';
  /** désactive la fermeture lors du clic sur le fond */
  disableDismiss?: boolean;
  /** n'affiche pas le bouton Fermer */
  noCloseButton?: boolean;
  /** Ajoute un effet flouté sur l'overlay */
  backdropBlur?: boolean;
  /** z-index de la modale, est placé sur l'overlay */
  zIndex?: string | number;
  /** Id de test */
  dataTest?: string;
};

/*
 * Basic modal
 * If you provide isOpen you must add the corresponding setState function.
 */
export const Modal = ({
  render,
  children,
  title,
  subTitle,
  description,
  textAlign = 'center',
  openState,
  onClose,
  size = 'md',
  disableDismiss,
  noCloseButton,
  renderFooter,
  backdropBlur,
  zIndex = preset.theme.extend.zIndex.modal,
  dataTest = 'Modal',
}: ModalProps) => {
  const isControlled = !!openState;
  const [open, setOpen] = useState(false);

  const isOpen = isControlled ? openState.isOpen : open;

  const handleOpenChange = () => {
    if (isControlled) {
      openState.setIsOpen(!openState.isOpen);
      openState.isOpen && onClose && onClose();
    } else {
      setOpen(!open);
      open && onClose && onClose();
    }
  };

  const {refs, context} = useFloating({
    open: isOpen,
    onOpenChange: handleOpenChange,
  });

  const id = useId();
  const labelId = `${id}-label`;
  const descriptionId = `${id}-description`;

  const {getReferenceProps, getFloatingProps} = useInteractions([
    useClick(context),
    useRole(context),
    useDismiss(context, {enabled: !disableDismiss}),
  ]);

  return (
    <>
      {children &&
        cloneElement(
          children,
          getReferenceProps({ref: refs.setReference, ...children.props})
        )}
      <FloatingPortal>
        {isOpen && (
          <FloatingOverlay
            lockScroll
            style={{
              display: 'grid',
              placeItems: 'center',
              background: preset.theme.extend.colors.overlay,
              zIndex,
              backdropFilter: backdropBlur ? 'blur(10px)' : undefined,
            }}
          >
            <FloatingFocusManager context={context}>
              <div
                data-test={dataTest}
                {...getFloatingProps({
                  ref: refs.setFloating,
                  'aria-labelledby': labelId,
                  'aria-describedby': descriptionId,
                  className: classNames(
                    `
                      relative flex flex-col self-end gap-8 w-full mt-8 mx-auto px-10 py-12
                      rounded-xl bg-white border border-grey-4 shadow-[0_4px_20px_0px_rgba(0,0,0,0.05)]
                      sm:self-center sm:w-[calc(100%-3rem)] sm:my-6
                      md:p-[4.5rem]
                    `,
                    sizeToClass[size]
                  ),
                })}
              >
                {!noCloseButton && (
                  <Button
                    data-test={`close-${dataTest}`}
                    title="Fermer"
                    onClick={handleOpenChange}
                    icon="close-line"
                    variant="grey"
                    size="xs"
                    className="absolute max-md:top-4 top-8 max-md:right-4 right-8"
                  />
                )}
                {(title || subTitle || description) && (
                  <div
                    className={classNames('flex flex-col gap-4', {
                      'text-left': textAlign === 'left',
                      'text-center': textAlign === 'center',
                      'text-right': textAlign === 'right',
                    })}
                  >
                    {title && (
                      <h3 id={labelId} className="mb-0 text-primary-10">
                        {title}
                      </h3>
                    )}
                    {subTitle && (
                      <p className="mb-0 font-medium text-grey-6">{subTitle}</p>
                    )}
                    {description && (
                      <p id={descriptionId} className="mb-0">
                        {description}
                      </p>
                    )}
                  </div>
                )}
                {render?.({
                  close: handleOpenChange,
                  labelId,
                  descriptionId,
                })}
                {renderFooter?.({
                  close: handleOpenChange,
                })}
              </div>
            </FloatingFocusManager>
          </FloatingOverlay>
        )}
      </FloatingPortal>
    </>
  );
};
