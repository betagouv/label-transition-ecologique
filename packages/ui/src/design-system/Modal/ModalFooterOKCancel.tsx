import {Button} from '@design-system/Button';
import {ButtonProps} from '../Button/types';
import {ModalFooter} from './ModalFooter';

/**
 * Variante de `ModalFooter` pour le cas courant "Annuler/Valider"
 */
export const ModalFooterOKCancel = (props: {
  /** Les props du bouton "Valider" */
  btnOKProps: ButtonProps;
  /** Les props du bouton "Annuler" (si non spécifié, le bouton est masqué) */
  btnCancelProps?: ButtonProps;
}) => {
  const {btnOKProps, btnCancelProps} = props;
  const {children: ok, ...btnOKRemainingProps} = btnOKProps;
  const {children: cancel, ...btnCancelRemainingProps} = btnCancelProps || {};

  return (
    <ModalFooter variant="right">
      {btnCancelProps && (
        <Button type="button" variant="outlined" {...btnCancelRemainingProps}>
          {cancel || 'Annuler'}
        </Button>
      )}
      <Button type="submit" {...btnOKRemainingProps}>
        {ok || 'Valider'}
      </Button>
    </ModalFooter>
  );
};
