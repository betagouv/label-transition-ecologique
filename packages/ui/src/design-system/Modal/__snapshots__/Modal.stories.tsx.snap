// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/Modal AvecBackdropBlur smoke-test 1`] = `
<div class="p-8">
  <button text="Ouvrir la modale"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
  >
    Ouvrir la modale
  </button>
</div>
`;

exports[`design-system/Modal AvecFooter smoke-test 1`] = `
<div class="p-8">
  <button text="Ouvrir la modale"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
  >
    Ouvrir la modale
  </button>
</div>
`;

exports[`design-system/Modal Controlled smoke-test 1`] = `
<div class="p-8">
  <button class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6">
    Open
  </button>
</div>
`;

exports[`design-system/Modal Default smoke-test 1`] = `
<div class="p-8">
  <button text="Ouvrir la modale"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
  >
    Ouvrir la modale
  </button>
</div>
`;

exports[`design-system/Modal LongContent smoke-test 1`] = `
<div class="p-8">
  <button text="Ouvrir la modale"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
  >
    Ouvrir la modale
  </button>
</div>
`;

exports[`design-system/Modal Size smoke-test 1`] = `
<div class="p-8">
  <div class="flex items-center gap-6">
    <button text="sm"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
    >
      sm
    </button>
    <button text="md"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
    >
      md
    </button>
    <button text="lg"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
    >
      lg
    </button>
    <button text="xl"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
    >
      xl
    </button>
  </div>
</div>
`;

exports[`design-system/Modal WithRender smoke-test 1`] = `
<div class="p-8">
  <button text="Ouvrir la modale"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
  >
    Ouvrir la modale
  </button>
</div>
`;

exports[`design-system/Modal onlyRender smoke-test 1`] = `
<div class="p-8">
  <button text="Ouvrir la modale"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="w-fit flex items-center border-solid group gap-2 cursor-pointer rounded-lg border font-bold text-primary-7 hover:!text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8 text-base py-3 px-6"
  >
    Ouvrir la modale
  </button>
</div>
`;
